src/main/java/hudson/plugins/git/extensions/impl/PreBuildMerge.java

```diff
@@ -85,7 +86,10 @@ public Revision decorateRevisionToBuild(GitSCM scm, Run<?, ?> build, GitClient g
             // record the fact that we've tried building 'rev' and it failed, or else
             // BuildChooser in future builds will pick up this same 'rev' again and we'll see the exact same merge failure
             // all over again.
-            scm.getBuildData(build).saveBuild(new Build(marked,rev, build.getNumber(), FAILURE));
+            BuildData bd = scm.getBuildData(build);
+            if(bd != null){
+                bd.saveBuild(new Build(marked,rev, build.getNumber(), FAILURE));
+            }
```

Comment:
If build data is missing, maybe makes sense to create it or to at least log the warning about unexpected behavior